resources/aws/subnet.go

```diff
@@ -100,17 +103,23 @@ func (s *Subnet) CreateOrFail() error {
 }
 
 func (s *Subnet) Delete() error {
```

Comment:
Why does a data structure has an API client? Data structures hold information. Services, managers, providers, etc. implement business logic. This is why we talked about the interface and I was under the impression we agreed on this. So in my opinion a good design would be to have a data structure holding information like some ID. This data structure or the plain ID can be provided to some service that implements business logic like deletion. The service has API clients configured. The service's `Delete` method expects at least one argument, the data structure or the ID. 

Since this here takes already longer than expected I would say lets go with this and do further improvements later on. Lets make sure the more important things get addressed.